src/cli/command_info.rs

```diff
@@ -4,7 +4,100 @@ use miette::{Context, IntoDiagnostic};
+use super::{Args, Command, get_styles};
+
+/// Get all built-in command names including aliases (discovered dynamically from clap)
+fn get_builtin_commands_with_aliases() -> Vec<String> {
+    let mut commands = Vec::new();
+
+    for subcommand in Args::command().get_subcommands() {
+        // Add main command name
+        commands.push(subcommand.get_name().to_string());
+
+        // Add all aliases
+        commands.extend(subcommand.get_all_aliases().map(|alias| alias.to_string()));
+    }
+
+    commands
+}
+
+/// Get all external command names (discovered from PATH)
+fn get_external_command_names() -> Vec<String> {
+    find_external_commands()
+        .into_iter()
+        .map(|(name, _path)| name)
+        .collect()
+}
+
+/// All available commands (built-in + external)
+fn get_all_available_commands() -> Vec<String> {
+    let mut all_commands = Vec::new();
+
+    all_commands.extend(get_builtin_commands_with_aliases());
+
+    all_commands.extend(get_external_command_names());
+
+    all_commands.sort();
+    all_commands.dedup();
+
+    all_commands
+}
+
+/// Find similar commands using Jaro similarity
+fn find_similar_commands(input: &str, threshold: f64) -> Vec<String> {
+    let available_commands = get_all_available_commands();
+    let mut suggestions: Vec<(f64, String)> = Vec::new();
+
+    for command in available_commands {
+        let similarity = strsim::jaro(input, &command);
+        if similarity > threshold {
+            suggestions.push((similarity, command));
+        }
+    }
+
+    // Sort by similarity (ascending), take top 3, reverse to get most similar first
+    suggestions.sort_by(|a, b| a.0.partial_cmp(&b.0).unwrap_or(std::cmp::Ordering::Equal));
+    suggestions
+        .into_iter()
+        .rev()
+        .take(3)
+        .map(|(_, cmd)| cmd)
+        .collect()
+}
+
+/// Find all external commands available in PATH
+fn find_external_commands() -> Vec<(String, PathBuf)> {
+    let mut commands = Vec::new();
+
+    if let Some(dirs) = search_directories() {
+        for dir in dirs {
+            if let Ok(entries) = fs_err::read_dir(&dir) {
+                for entry in entries.flatten() {
+                    if let Some(name) = entry.file_name().to_str() {
+                        // Check if it's a pixi extension
+                        if let Some(cmd_name) = name.strip_prefix("pixi-") {
+                            // Remove .exe suffix on Windows
+                            let cmd_name = cmd_name
+                                .strip_suffix(env::consts::EXE_SUFFIX)
+                                .unwrap_or(cmd_name);
+
+                            let path = entry.path();
+                            if path.is_executable() {
+                                commands.push((cmd_name.to_string(), path));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Remove duplicates
```

Comment:
lets use a hashmap for this if we still store name - path. it is more appropriate and it automatically can dedup it